testdata/project-v4-with-plugins/internal/controller/memcached_controller.go

```diff
@@ -190,23 +179,47 @@ func (r *MemcachedReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 		return ctrl.Result{}, nil
 	}
 
+	// Let's add a finalizer. Then, we can define some operations which should
+	// occur before the custom resource is deleted.
+	// More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/finalizers
+	if !controllerutil.ContainsFinalizer(memcached, memcachedFinalizer) {
+		log.Info("Adding Finalizer for Memcached")
+		if ok := controllerutil.AddFinalizer(memcached, memcachedFinalizer); !ok {
+			log.Error(err, "Failed to add finalizer into the custom resource")
+			return ctrl.Result{}, fmt.Errorf("Failed to add finalizer into the custom resource")
+		}
+
+		if err = r.Update(ctx, memcached); err != nil {
+			log.Error(err, "Failed to update custom resource to add finalizer")
+			return ctrl.Result{}, err
+		}
+
+		// TODO GET call?
+	}
+
 	// Check if the deployment already exists, if not create a new one
 	found := &appsv1.Deployment{}
 	err = r.Get(ctx, types.NamespacedName{Name: memcached.Name, Namespace: memcached.Namespace}, found)
-	if err != nil && apierrors.IsNotFound(err) {
+	if err != nil {
```

Comment:
Why do this change?

Reply:
I think it easier to read

it was
```go
	if err != nil && apierrors.IsNotFound(err) {
		// create a new deployment
	} else if err != nil {
		// error
	}
```

my suggestion is
```go
	if err != nil {
		if !apierrors.IsNotFound(err) {
			// error
		}
		// create a new deployment
	}
```